python/objToJSON.c

```diff
@@ -579,7 +592,7 @@ void Object_beginTypeContext (JSOBJ _obj, JSONTypeContext *tc)
     return;
   }
   else
-  if (PyString_Check(obj))
+  if (PyString_Check(obj) && !PyObject_HasAttrString(obj, "__json__"))
```

Comment:
Thanks for the explanation. However this is too much of an edge-case to justify the performance penalty as it stands.